### ISSUE TITLE
`term_start` can't redirect input from file

### ISSUE THREAD

**Issue by senioria** [vim/vim#3593]
My English is poor, so if I make any mistake, please forgive me.

I made a plugin to compile and run c++ source file, and I found that it doesn't work when I attemp to redirect the executable's input from a file. I tried with `vim -u NONE -N`, but it still didn't work.

The mininum code that can reproduce this issue:
```vim
call term_start('cat.exe', {'in_io': 'file', 'in_name': 'input.txt'})
```
In which `cat.exe` acts like `cat` on *nix.

I'm using Windows, and I didn't test this on other platform, so I'm not sure if this is caused by Windows.

**Comment by brammool** [comment]
> My English is poor, so if I make any mistake, please forgive me.
> 
> I made a plugin to compile and run c++ source file, and I found that it doesn't work when I attemp to redirect the executable's input from a file. I tried with `vim -u NONE -N`, but it still didn't work.
> 
> The mininum code that can recure this issue:
> ```vim
> call term_start('cat.exe', {'in_io': 'file', 'in_name': 'input.txt'})
> ```
> In which `cat.exe` acts like `cat` on *nix.
> 
> I'm using Windows, and I didn't test this on other platform, so I'm
> not sure if this is caused by Windows.

Yes, this is caused by windows.  The WinPTY library used does not
support redirection.  You will have to find another way to do this.

-- 
What the word 'politics' means: 'Poli' in Latin meaning 'many' and 'tics'
meaning 'bloodsucking creatures'.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by lacygoill** [comment]
@91khr: I don't know much about Windows, but it seems that since [8.1.0870](https://github.com/vim/vim/releases/tag/v8.1.0870), Vim supports an alternative to winpty: [ConPTY](https://vimhelp.org/terminal.txt.html#ConPTY).  Does it help?  I'm not sure but you might need to set the global option `'termwintype'` to `conpty`.
